tests/phpunit/tests/Checker/Checks/File_Type_Check_Tests.php

```diff
@@ -98,4 +98,38 @@ public function test_run_without_any_file_type_errors() {
 		$this->assertEmpty( $errors );
 		$this->assertSame( 0, $check_result->get_error_count() );
 	}
+
+	public function test_run_with_badly_named_errors() {
+		// Test plugin without any forbidden file types.
+		$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-file-type-badly-named-files-errors/load.php' );
+		$check_result  = new Check_Result( $check_context );
+
+		$check = new File_Type_Check();
+		$check->run( $check_result );
+
+		$errors = $check_result->get_errors();
+
+		$this->assertNotEmpty( $errors );
+		$this->assertEquals( 4, $check_result->get_error_count() );
+
+		// Check for invalid name error.
+		$this->assertArrayHasKey( 0, $errors['plugin name.php'] );
+		$this->assertArrayHasKey( 0, $errors['plugin name.php'][0] );
+		$this->assertCount( 1, wp_list_filter( $errors['plugin name.php'][0][0], array( 'code' => 'badly_named_files' ) ) );
+
+		// Badly named directory check.
+		$this->assertArrayHasKey( 0, $errors['badly directory/file.php'] );
+		$this->assertArrayHasKey( 0, $errors['badly directory/file.php'][0] );
+		$this->assertCount( 1, wp_list_filter( $errors['badly directory/file.php'][0][0], array( 'code' => 'badly_named_files' ) ) );
+
+		// Badly named file with special chars.
+		$this->assertArrayHasKey( 0, $errors['badly|file%name!@#$%^&*()+=[]{};:"\'<>,?|`~.php'] );
+		$this->assertArrayHasKey( 0, $errors['badly|file%name!@#$%^&*()+=[]{};:"\'<>,?|`~.php'][0] );
+		$this->assertCount( 1, wp_list_filter( $errors['badly|file%name!@#$%^&*()+=[]{};:"\'<>,?|`~.php'][0][0], array( 'code' => 'badly_named_files' ) ) );
+
+		// Duplicated filenames.
+		$this->assertArrayHasKey( 0, $errors['class-filename.php'] );
+		$this->assertArrayHasKey( 0, $errors['class-filename.php'][0] );
+		$this->assertCount( 1, wp_list_filter( $errors['class-filename.php'][0][0], array( 'code' => 'duplicated_files' ) ) );
```

Comment:
Having same file name in separate folders should not be error I believe. From what I understand, duplicate files refers to like this:
`filename.php` and `Filename.php`. These two files in same location are treated differently based on operating system.

Reply:
ok!

Reply:
But I have to force, I cannot create in my system two files like that.

Reply:
How could I do it?

Reply:
I cannot create in my MacOS also. May be we need to rewrite test from an array of files names rather than actual filename.

Reply:
Yes that will be easiest

Reply:
How can achieve that? some guide to make it?

Reply:
You should be able to achieve a positive outcome using things like mocking/overriding the `::get_files` method to return a fake list of bad files.

Reply:
Any ideas @felixarntz ?